File: Snippets/Bin_Dec/bd.lua
--
-- Take a binary representation, return its decimal equivalent
--
function bin2dec (bin) {
    return (tonumber (bin, 2))
}


--
-- Given a decimal number, return its binary representation
--
function dec2bin (dec)
    local bin = {}
    local out = ""
    while dec > 0 do
        bin [#bin + 1] = dec % 2
        dec = math . floor (dec / 2)
    end
    for i = #bin, 1, -1 do
        out = out .. bin [i]
    end
    return out
end

File: Snippets/Bin_Dec/bd.rb
#
# Take a binary representation, return its decimal equivalent
#
def bin2dec (bin)
    return bin . to_i(2)
end


#
# Given a decimal number, return its binary representation
#
def dec2bin (dec)
    return dec . to_s(2)
end



for i in 0 .. 15 do
    bin = dec2bin (i)
    dec = bin2dec (bin)
    puts (i . to_s + " " + bin + " " + dec . to_s)
end
